catalogue_graph/tests/adapters/ebsco/test_loader.py

```diff
@@ -1,17 +1,59 @@
 import json
 
 import pytest
+from lxml import etree
 from pyiceberg.table import Table as IcebergTable
 
 from adapters.ebsco.models.step_events import (
     EbscoAdapterLoaderEvent,
     EbscoAdapterTransformerEvent,
 )
-from adapters.ebsco.steps.loader import EbscoAdapterLoaderConfig, handler
+from adapters.ebsco.steps.loader import (
+    EbscoAdapterLoaderConfig,
+    extract_record_id,
+    handler,
+)
 from adapters.ebsco.utils.tracking import ProcessedFileRecord
 from tests.mocks import MockSmartOpen
 
 
+class TestRecordIdExtraction:
+    def test_uses_controlfield_001_when_available(self) -> None:
+        node = etree.fromstring(
+            """
+            <record>
+                <controlfield tag="001">r-control</controlfield>
+                <datafield tag="035">
+                    <subfield code="a">(OCoLC)9999999</subfield>
+                </datafield>
+            </record>
+            """
+        )
+
+        assert extract_record_id(node) == "r-control"
+
+    def test_falls_back_to_datafield_035_removing_prefix(self) -> None:
+        node = etree.fromstring(
+            """
+            <record>
+                <datafield tag="035">
+                    <subfield code="a">(OCoLC)814782</subfield>
+                </datafield>
+            </record>
+            """
+        )
+
+        assert extract_record_id(node) == "814782"
+
+    def test_raises_when_no_identifier_present(self) -> None:
```

Comment:
Some other scenarios to exercise - when the fields are present but empty or invalid.

```suggestion
    def test_ignores_controlfield_001_when_empty(self) -> None:
        node = etree.fromstring(
            """
            <record>
                <controlfield tag="001"></controlfield>
                <datafield tag="035">
                    <subfield code="a">(OCoLC)9999999</subfield>
                </datafield>
            </record>
            """
        )
        assert extract_record_id(node) == "9999999"

    def test_raises_when_no_identifier_present(self) -> None:

        with pytest.raises(
                Exception, match="Could not find controlfield 001 or usable datafield 035"
        ):
            node = etree.fromstring(
                """
                <record>
                    <controlfield tag="001">   </controlfield>
                    <datafield tag="035">
                        <subfield code="a"> (OCoLC) </subfield>
                    </datafield>
                </record>
                """
            )
        assert extract_record_id(node)

    def test_raises_when_no_identifier_fields_present(self) -> None:
```